### ISSUE TITLE
RabbitMQ alternating error "There is no equivalent message pattern defined in the remote service"

### ISSUE THREAD

**Issue by KhaustovN1ck** [nestjs/nest#1492]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

I have a few microservices with all the necessary business logic and message patterns. When I try to call one of them via `que.send({cmd: [MyPattern]})` first call always fails (and all the odd calls respectively) , second (and all the even calls respectively) always succeed.
related: https://github.com/nestjs/nest/issues/1245

## Minimal reproduction of the problem with instructions
Simplified Description:
 I have 2 microservices, first one is configured like this

```typescript
// main.ts
 app.connectMicroservice({
        transport: Transport.RMQ,
        options: {
            urls: [`amqp://localhost:5672`],
            queue: 'manager_service_queue',
            queueOptions: {
                durable: false,
           }
      }
});
```
and has controller like this:

```typescript
@Controller()
export class MyController {

    constructor(
        private ordersService: OrdersService,
        private logger: AppLoggerService){}

    @MessagePattern({cmd: RmqPatterns.ORDER_PAYMENT_REJECTED})
    async orderRejected(paymentStatusDto: PaymentStatusDto) {
        this.ordersService.changeOrderStatus(
            // ... DTO
        );
    }

    @MessagePattern({cmd: RmqPatterns.ORDER_CREATED})
    async orderCreated(newOrder: Order) {
        this.ordersService.notifyNewOrder(newOrder);
    }

    @MessagePattern({cmd: RmqPatterns.ORDER_PAYMENT_APPROVED})
    async orderApproved(paymentStatusDto: PaymentStatusDto) {
        this.ordersService.changeOrderStatus(
            // DTO
        );
    }
}
```
second one has this logic:
```typescript
// my-service.service.ts
// ...
 @Client({
    transport: Transport.RMQ,
    options: {
      urls: [`amqp://localhost:5672`],
      queue: 'manager_service_queue',
      queueOptions: {durable: false},
    },
  })
  managerQueue: ClientProxy;

  callAnotherService(){
     return this.managerQueue.send({cmd: RmqPatterns.ORDER_CREATED}, order);
  }
// ....
```
When callAnotherService method is called first time I receive this error:
*There is no equivalent message pattern defined in the remote service*
Second call seems to be OK, third fails again, fourth work, etc. 😄
In other words even calls work and odd calls fail.
Does anyone know why it happens? And perhaps more important thing is what to do with that 🤔
Will appreciate any help 🙏

## Environment

<pre><code>
Nest version: 5.4.0

 
For Tooling issues:
- Node version: 11.0.0
- Platform:  MacOS

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Could you provide a minimal repository that reproduces your issue? I'll clone it locally and test against my local broker.

**Comment by KhaustovN1ck** [comment]
@kamilmysliwiec I've created a repo reproducing this issue with all the comments in README.md - https://github.com/KhaustovN1ck/RabbitMQ-Nest-Error. Thanks in advance for your help 😃

**Comment by cojack** [comment]
@kamilmysliwiec can we make a `names` for transport clients? It will be much easier in the `controllers` to provide just name of the client, instead of whole configuration, like we do in the `connectMicroservice` method.

**Comment by patricknazar** [comment]
Related/duplicate #1245

**Comment by IllyaHorbatenko** [comment]
Happens to me too, any ideas of what to do with it?

**Comment by patricknazar** [comment]
@KhaustovN1ck I've played with your repo. [this line](https://github.com/KhaustovN1ck/RabbitMQ-Nest-Error/blob/master/manager-app/src/main.ts#L13) should be `manager_service_queue` instead of `payment_service_queue` I believe. Seems to work with that change.

**Comment by kamilmysliwiec** [comment]
Thanks @patricknazar!
